src/Integration/LocalServices/ProjectSystemFilter.cs

```diff
@@ -85,13 +72,58 @@ public bool IsAccepted(DteProject dteProject)
 
         public void SetTestRegex(Regex regex)
         {
+            if (regex == null)
+            {
+                throw new ArgumentNullException(nameof(regex));
+            }
+
             this.testRegex = regex;
-            Debug.Assert(this.testRegex == null || (this.testRegex.MatchTimeout != Regex.InfiniteMatchTimeout), "Should have set non-infinite timeout");
+            Debug.Assert(this.testRegex.MatchTimeout != Regex.InfiniteMatchTimeout, "Should have set non-infinite timeout");
         }
 
         #endregion
 
         #region Helpers
+        private static bool IsTestProject(IVsHierarchy projectHierarchy, Regex testProjectNameRegex, string projectName)
+        {
+            // Otherwise, try to detect test project using known project types and/or regex match
```

Comment:
This comment is copy-paste same from line 64. Should be reworded; "otherwise" was there because of an if-else.